### PR TITLE
SSSOM gene mapping examples

### PR DESCRIPTION
**Pull request by patrick-lloyd-ray**
@satra @lydiang 

This is done with SSSOM, just showing functionality in two forms: one across species, one within. For revision, we'd use within and simply specify the IDs to tie them to a specific revision (even though my example maps ensembl to ncbigene, it's just to illustrate). 

Happy to iterate, tie into LinkML models, experiment with how to make this work in our ecosystem once we get the elements correct.

### CODE REVIEW

File: docs/mappings/gene_mapping_human.tsv [hunk 1]
```diff
@@ -0,0 +1,2 @@
+subject_id	subject_label	predicate_id	object_id	object_label	mapping_justification	mapping_date	author_id	subject_source	subject_source_version	object_source	object_source_version	confidence
```
Comment (satra) [inline docs/mappings/gene_mapping_human.tsv:1]: are all of these headings from sssom?

**Review by satra — COMMENTED** [review]
i left one question. 

the main question i would have is do you see a way to automate this from a mapping released by ensembl/ncbigene? as in, can one write a function or transform?

File: docs/mappings/gene_mapping_human.tsv (hunk 1 above)
Comment (patrick-lloyd-ray) [reply docs/mappings/gene_mapping_human.tsv:1]: Yes, there are up to 30 metadata elements we can include here. I just picked a standard set as an example.

**Comment by patrick-lloyd-ray** [comment]
> i left one question.
> 
> the main question i would have is do you see a way to automate this from a mapping released by ensembl/ncbigene? as in, can one write a function or transform?

This would be tremendously helpful.

From what I've seen, there are some tools that will query these databases (https://github.com/pachterlab/gget is an example) but I haven't seen a way to automate this from a mapping released by ensembl/ncbigene. 

I will look into this more deeply and give an update in the next 24 hours.

**Comment by lydiang** [comment]
Regarding mapping.

We need to make sure we have good provenance of this. 
At each annotation release for Ensembl and NCBI, the also release mappings (NCBI-Ensembl) and orthologs.

There is no guarantee that mappings from each authority is identical. We should ingest mapping like we do the annotation - eg with version and source etc.

**Comment by patrick-lloyd-ray** [comment]
> Regarding mapping.
> 
> We need to make sure we have good provenance of this. At each annotation release for Ensembl and NCBI, the also release mappings (NCBI-Ensembl) and orthologs.
> 
> There is no guarantee that mappings from each authority is identical. We should ingest mapping like we do the annotation - eg with version and source etc.

The metadata fields `subject_source_version` and `object_source_version` are used to track versions in this model. We can add other fields, if that's not sufficient.

**Comment by patrick-lloyd-ray** [comment]
@satra I can't find any existing tooling that would automate this completely. 

There are things like: [GIDcon](http://resource.ibab.ac.in/GIDCON/geneid/home.html) and [biomart]( https://www.ensembl.org/biomart/martview/22dd85b61ca0211407ea33805571076f) that can produce tables that will map IDs, but we'd have to curate them manually here with metadata. We could schedule a quarterly release/update, though, if it's not a ton of work.

**Comment by satra** [comment]
@sooyounga and i can figure out how to parse and write as long as there is a simple enough mapping. would it be possible for you to let @sooyounga know which source fields you put into the sssom fields?

**Comment by patrick-lloyd-ray** [comment]
> @sooyounga and i can figure out how to parse and write as long as there is a simple enough mapping. would it be possible for you to let @sooyounga know which source fields you put into the sssom fields?

Yes, the source fields are listed in the table. 

I also came across this tool (https://biit.cs.ut.ee/gprofiler/convert) that may be useful as well.

**Pull request merged**